python/tvm/meta_schedule/tune_context.py

```diff
@@ -179,6 +214,8 @@ def pre_tuning(
                 "search_strategy is not provided."
                 "Please construct TuneContext with search_strategy"
             )
+        if design_spaces is None:
+            design_spaces = self.generate_design_space()
         return self.search_strategy.pre_tuning(design_spaces, database, cost_model)
```

Comment:
We could add a case like "if the search strategy is evolutionary search, and database and cost_model are none, we automatically use `MemoryDatabase` and `RandomModel`".

Reply:
Good idea!! Let me tweak it accordingly